h2o-core/src/main/java/water/util/VecUtils.java

```diff
@@ -838,4 +838,66 @@ public void map(Chunk c, NewChunk nc) {
     }
   }
 
+  /**
+   * shuffle values of a Vec {@link Vec}.
+   * 
+   * The 'new' values of the vector will be the already exisitng one's however randomly shuffeled
+   * 
+   * 
+   */
+  
+  public static class shuffleVec extends MRTask<shuffleVec>{
```

Comment:
Why are you have two implementations of ShuffleVec? VecUtils is probably a better place to place the code.

Reply:
I would call it brainstorming :)

Yes upon refactoring that's where I will put it. I had an Idea to shuffle the whole training data Frame as "pre-processing" therefore I wanted to store the Shuffled Frame on a class and then only use swapping methods between two frames. However, not pursuing that idea.